src/charm.py

```diff
@@ -1,103 +1,217 @@
 #!/usr/bin/env python3
-# Copyright 2024 Ali
+# Copyright 2024 Canonical Ltd.
 # See LICENSE file for licensing details.
-#
-# Learn more at: https://juju.is/docs/sdk
 
-"""Charm the service.
-
-Refer to the following tutorial that will help you
-develop a new k8s charm using the Operator Framework:
-
-https://juju.is/docs/sdk/create-a-minimal-kubernetes-charm
-"""
+"""Charm the application."""
 
 import logging
+from pathlib import Path
 
 import ops
+from charms.data_platform_libs.v0.data_models import TypedCharmBase
+from log import log_event_handler
+from ops.model import ActiveStatus, BlockedStatus, WaitingStatus
+from state import State
+from structured_config import CharmConfig
 
-# Log messages can be retrieved using juju debug-log
 logger = logging.getLogger(__name__)
 
-VALID_LOG_LEVELS = ["info", "debug", "warning", "error", "critical"]
+CONTAINERS = [
+    "airbyte-api-server",
+    "airbyte-bootloader",
+    "airbyte-connector-builder-server",
+    "airbyte-cron",
+    "airbyte-pod-sweeper",
+    "airbyte-server",
+    "airbyte-worker",
+]
+CONNECTOR_BUILDER_SERVER_API_PORT = 80
+INTERNAL_API_PORT = 8001
+AIRBYTE_API_PORT = 8006
+WORKLOAD_API_PORT = 8007
+
+# TODO (kelkawi-a): perform up check on the following ports for each container
+# airbyte-api-server: 8006
+# airbyte-bootloader: None
+# airbyte-connector-builder-server: 8080
+# airbyte-cron: 9001
+# airbyte-pod-sweeper: None
+# airbyte-server: 8001
+# airbyte-worker: 9000
+
+
+def get_pebble_layer(application_name, context):
+    if application_name == "airbyte-worker":
+        application_name = "airbyte-workers"
+
+    command = f"/bin/bash -c airbyte-app/bin/{application_name}"
+    if application_name == "airbyte-pod-sweeper":
```

Comment:
question: What is the cause of this switch?

Reply:
This specific container uses the `bitnami/kubectl` OCI image, as opposed to the other containers which use OCI images from Airbyte with defined entry points. So for this one, we push a script that needs to be run as the entry point of this container, whose job is to sweep Airbyte job pods that have passed a certain TTL.

Reply:
Not the command, I meant the change in `application_name`.

Reply:
Ah, for this specific container, the name of the container (`airbyte-worker`) differs from the name in the entry point (`airbyte-workers`).

Reply:
can't you use the entrypoint name as a container name?

Reply:
Good point, updated.